packages/data-broker/src/DataBroker.js

```diff
@@ -85,33 +88,66 @@ export class DataBroker {
     return syncGroups.map(sg => ({ ...sg, type: this.getDataSourceTypes().SYNC_GROUP }));
   };
 
-  checkDataElementPermissions = async dataElements => {
-    const userPermissions = await this.getUserPermissions();
-    if (userPermissions.includes(BES_ADMIN_PERMISSION_GROUP)) {
+  getCountryCodes = async organisationUnitCodes => {
+    const orgUnits = await this.models.entity.find({ code: organisationUnitCodes });
+    const orgUnitCountryCodes = orgUnits
+      .map(orgUnit => orgUnit.country_code)
+      .filter(countryCode => countryCode !== null && countryCode !== undefined);
+    const countryCodes = [...new Set(orgUnitCountryCodes)];
+    return countryCodes;
+  };
+
+  checkDataElementPermissions = async (dataElements, countryCodes) => {
+    const allUserPermissions = this.getUserPermissions();
+    if (allUserPermissions.includes(BES_ADMIN_PERMISSION_GROUP)) {
       return true;
     }
-    const missingPermissions = [];
-    for (const element of dataElements) {
-      if (
-        element.permission_groups.length <= 0 ||
-        element.permission_groups.some(code => userPermissions.includes(code))
-      ) {
-        continue;
+
+    const getDataElementsWithMissingPermissions = permissions =>
+      dataElements
+        .filter(element => element.permission_groups.length > 0)
+        .filter(element => !element.permission_groups.some(group => permissions.includes(group)))
+        .map(element => element.code);
+
+    if (countryCodes === null) {
```

Comment:
It's possible for countryCode to be null when fetching for `sync_groups` I believe. At the moment sync groups don't do permissions checks, but they could... so leaving this here for now 👍